### PR TITLE
fix(rolling upgrade file): fit it branch-2020.1

### PR DESCRIPTION
**Pull request by fgelcer**
the file was created on master, and once backported
to branch-2020.1, it didn't work.
here fixing it to make it work.

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Review by amoskong — APPROVED** [review]
LGTM

File: sdcm/utils/ldap.py [hunk 1]
```diff
@@ -58,6 +58,8 @@ def is_ldap_connection_bound():
     @staticmethod
     def add_ldap_entry(ip, ldap_port, user, password, ldap_entry):
         if not LdapContainerMixin.is_ldap_connection_bound():
+            sleep(5)
+            print('this is creating a ldap_connection with values {}'.format([ip, ldap_port, user, password]))
```
Comment (amoskong) [inline sdcm/utils/ldap.py:62]: Can you switch to logging?

**Review by fruch — APPROVED** [review]
LGTM

But can you take a look at the pylint issues ? some of them sounds problematic

File: sdcm/utils/ldap.py (hunk 1 above)
Comment (fruch) [reply sdcm/utils/ldap.py:62]: Also can you add comment to explain this sleep ?

And if important add it also in master, cause soon we'll branch 2021.1...

Comment (fgelcer) [reply sdcm/utils/ldap.py:62]: switched, @amoskong .
@fruch , on previous tries I didn't see the need for it, but as this one was failing, I added as debug, and altogether with other changes, it won't bother, and if we find that we want to add this further, i will then send a PR to master and branch-2021.1

**Comment by fgelcer** [comment]
> LGTM
> 
> But can you take a look at the pylint issues ? some of them sounds problematic

i'm searching for pylint issues, but can't find them

**Comment by fgelcer** [comment]
> > LGTM
> > But can you take a look at the pylint issues ? some of them sounds problematic
> 
> i'm searching for pylint issues, but can't find them

@fruch , found when running locally `pre-commit run --all-files`:
```
************* Module mgmt_cli_test
mgmt_cli_test.py:385:22: E1123: Unexpected keyword argument 'single_node' in method call (unexpected-keyword-arg)
************* Module query_limits_test
query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
************* Module cdc_replication_test
cdc_replication_test.py:23:0: E0611: No name 'SimpleStatement' in module 'cassandra.query' (no-name-in-module)
query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
cdc_replication_test.py:23:0: E0611: No name 'SimpleStatement' in module 'cassandra.query' (no-name-in-module)
************* Module unit_tests.test_ycsb_thread
unit_tests/test_ycsb_thread.py:40:4: E0611: No name 'Cluster' in module 'cassandra.cluster' (no-name-in-module)
query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
cdc_replication_test.py:23:0: E0611: No name 'SimpleStatement' in module 'cassandra.query' (no-name-in-module)
unit_tests/test_ycsb_thread.py:40:4: E0611: No name 'Cluster' in module 'cassandra.cluster' (no-name-in-module)
************* Module sdcm.cluster_aws
sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'ec2_user_data' in method call (unexpected-keyword-arg)
sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'cluster_uuid' in method call (unexpected-keyword-arg)
sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'cluster_prefix' in method call (unexpected-keyword-arg)
sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'node_prefix' in method call (unexpected-keyword-arg)
sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'node_type' in method call (unexpected-keyword-arg)
sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'ec2_user_data' in method call (unexpected-keyword-arg)
sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'cluster_uuid' in method call (unexpected-keyword-arg)
sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'cluster_prefix' in method call (unexpected-keyword-arg)
sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'node_prefix' in method call (unexpected-keyword-arg)
sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'node_type' in method call (unexpected-keyword-arg)
************* Module sdcm.fill_db_data
sdcm/fill_db_data.py:29:0: E0611: No name 'ProtocolException' in module 'cassandra.protocol' (no-name-in-module)
sdcm/fill_db_data.py:29:0: E0611: No name 'ProtocolException' in module 'cassandra.protocol' (no-name-in-module)
sdcm/fill_db_data.py:29:0: E0611: No name 'ProtocolException' in module 'cassandra.protocol' (no-name-in-module)
```

none of these files i touched, but still i can try to fix them..

**Comment by fruch** [comment]
> > > LGTM
> > > But can you take a look at the pylint issues ? some of them sounds problematic
> > 
> > 
> > i'm searching for pylint issues, but can't find them
> 
> @fruch , found when running locally `pre-commit run --all-files`:
> 
> ```
> ************* Module mgmt_cli_test
> mgmt_cli_test.py:385:22: E1123: Unexpected keyword argument 'single_node' in method call (unexpected-keyword-arg)
> ************* Module query_limits_test
> query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
> ************* Module cdc_replication_test
> cdc_replication_test.py:23:0: E0611: No name 'SimpleStatement' in module 'cassandra.query' (no-name-in-module)
> query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
> cdc_replication_test.py:23:0: E0611: No name 'SimpleStatement' in module 'cassandra.query' (no-name-in-module)
> ************* Module unit_tests.test_ycsb_thread
> unit_tests/test_ycsb_thread.py:40:4: E0611: No name 'Cluster' in module 'cassandra.cluster' (no-name-in-module)
> query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
> query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
> query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
> query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
> query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
> cdc_replication_test.py:23:0: E0611: No name 'SimpleStatement' in module 'cassandra.query' (no-name-in-module)
> unit_tests/test_ycsb_thread.py:40:4: E0611: No name 'Cluster' in module 'cassandra.cluster' (no-name-in-module)
> ************* Module sdcm.cluster_aws
> sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'ec2_user_data' in method call (unexpected-keyword-arg)
> sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'cluster_uuid' in method call (unexpected-keyword-arg)
> sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'cluster_prefix' in method call (unexpected-keyword-arg)
> sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'node_prefix' in method call (unexpected-keyword-arg)
> sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'node_type' in method call (unexpected-keyword-arg)
> sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'ec2_user_data' in method call (unexpected-keyword-arg)
> sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'cluster_uuid' in method call (unexpected-keyword-arg)
> sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'cluster_prefix' in method call (unexpected-keyword-arg)
> sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'node_prefix' in method call (unexpected-keyword-arg)
> sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'node_type' in method call (unexpected-keyword-arg)
> ************* Module sdcm.fill_db_data
> sdcm/fill_db_data.py:29:0: E0611: No name 'ProtocolException' in module 'cassandra.protocol' (no-name-in-module)
> sdcm/fill_db_data.py:29:0: E0611: No name 'ProtocolException' in module 'cassandra.protocol' (no-name-in-module)
> sdcm/fill_db_data.py:29:0: E0611: No name 'ProtocolException' in module 'cassandra.protocol' (no-name-in-module)
> ```
> 
> none of these files i touched, but still i can try to fix them..

it would be nice to handle them for this branch, and yeah we can do that in a different PR, it would help our visibility when doing PRs into this branch, to make sure we are not breaking anything too back with backports and such

**Review by fruch — APPROVED** [review]
LGTM

**Comment by fgelcer** [comment]
> > > > LGTM
> > > > But can you take a look at the pylint issues ? some of them sounds problematic
> > > 
> > > 
> > > i'm searching for pylint issues, but can't find them
> > 
> > 
> > @fruch , found when running locally `pre-commit run --all-files`:
> > ```
> > ************* Module mgmt_cli_test
> > mgmt_cli_test.py:385:22: E1123: Unexpected keyword argument 'single_node' in method call (unexpected-keyword-arg)
> > ************* Module query_limits_test
> > query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
> > ************* Module cdc_replication_test
> > cdc_replication_test.py:23:0: E0611: No name 'SimpleStatement' in module 'cassandra.query' (no-name-in-module)
> > query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
> > cdc_replication_test.py:23:0: E0611: No name 'SimpleStatement' in module 'cassandra.query' (no-name-in-module)
> > ************* Module unit_tests.test_ycsb_thread
> > unit_tests/test_ycsb_thread.py:40:4: E0611: No name 'Cluster' in module 'cassandra.cluster' (no-name-in-module)
> > query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
> > query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
> > query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
> > query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
> > query_limits_test.py:56:8: E1123: Unexpected keyword argument 'targets' in method call (unexpected-keyword-arg)
> > cdc_replication_test.py:23:0: E0611: No name 'SimpleStatement' in module 'cassandra.query' (no-name-in-module)
> > unit_tests/test_ycsb_thread.py:40:4: E0611: No name 'Cluster' in module 'cassandra.cluster' (no-name-in-module)
> > ************* Module sdcm.cluster_aws
> > sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'ec2_user_data' in method call (unexpected-keyword-arg)
> > sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'cluster_uuid' in method call (unexpected-keyword-arg)
> > sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'cluster_prefix' in method call (unexpected-keyword-arg)
> > sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'node_prefix' in method call (unexpected-keyword-arg)
> > sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'node_type' in method call (unexpected-keyword-arg)
> > sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'ec2_user_data' in method call (unexpected-keyword-arg)
> > sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'cluster_uuid' in method call (unexpected-keyword-arg)
> > sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'cluster_prefix' in method call (unexpected-keyword-arg)
> > sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'node_prefix' in method call (unexpected-keyword-arg)
> > sdcm/cluster_aws.py:890:8: E1123: Unexpected keyword argument 'node_type' in method call (unexpected-keyword-arg)
> > ************* Module sdcm.fill_db_data
> > sdcm/fill_db_data.py:29:0: E0611: No name 'ProtocolException' in module 'cassandra.protocol' (no-name-in-module)
> > sdcm/fill_db_data.py:29:0: E0611: No name 'ProtocolException' in module 'cassandra.protocol' (no-name-in-module)
> > sdcm/fill_db_data.py:29:0: E0611: No name 'ProtocolException' in module 'cassandra.protocol' (no-name-in-module)
> > ```
> > 
> > 
> > none of these files i touched, but still i can try to fix them..
> 
> it would be nice to handle them for this branch, and yeah we can do that in a different PR, it would help our visibility when doing PRs into this branch, to make sure we are not breaking anything too back with backports and such

opened a trello task to handle this (feel free to assign to someone, if before i arrive once again to it):
https://trello.com/c/kEwZ5GgR/2936-handle-and-fix-all-pylint-issues-on-branch-20201

**Pull request merged**